en/tutorials-and-examples/blog-auth-example/auth.rst

```diff
@@ -245,189 +153,205 @@ entity file and add the following::
+Test it by requesting any page of your site. After being redirected
+to the ``/users/login`` page, enter the email and password you
+picked previously when creating your user. You should be redirected
+successfully after login.
 
-        // The owner of an article can edit and delete it
-        if (in_array($this->request->getParam('action'), ['edit', 'delete'])) {
-            $articleId = (int)$this->request->getParam('pass.0');
-            if ($this->Articles->isOwnedBy($articleId, $user['id'])) {
-                return true;
-            }
-        }
+We need to add a couple more details to configure our application.  We want all
+``view`` and ``index`` pages accessible without logging in so we'll add this
+specific configuration in ``AppController``::
 
-        return parent::isAuthorized($user);
+    // in src/Controller/AppController.php
+    public function beforeFilter(\Cake\Event\EventInterface $event)
+    {
+        parent::beforeFilter($event);
+        // for all controllers in our application, make index and view
+        // actions public, skipping the authentication check.
+        $this->Authentication->addUnauthenticatedActions(['index', 'view']);
     }
 
-We're now overriding the AppController's ``isAuthorized()`` call and internally
-checking if the parent class is already authorizing the user. If he isn't,
-then just allow him to access the add action, and conditionally access
-edit and delete. One final thing has not been implemented. To tell whether
-or not the user is authorized to edit the article, we're calling a ``isOwnedBy()``
-function in the Articles table. Let's then implement that function::
+Logout
+======
 
-    // src/Model/Table/ArticlesTable.php
+Add the logout action to the ``UsersController`` class::
 
-    public function isOwnedBy($articleId, $userId)
+    // in src/Controller/UsersController.php
+    public function logout()
     {
-        return $this->exists(['id' => $articleId, 'user_id' => $userId]);
+        $result = $this->Authentication->getResult();
+        // regardless of POST or GET, redirect if user is logged in
+        if ($result->isValid()) {
+            $this->Authentication->logout();
+            return $this->redirect(['controller' => 'Users', 'action' => 'login']);
+        }
     }
 
-This concludes our simple authentication and authorization tutorial. For securing
-the UsersController you can follow the same technique we did for ArticlesController.
-You could also be more creative and code something more general in AppController based
-on your own rules.
-
-Should you need more control, we suggest you read the complete Auth guide in the
-:doc:`/controllers/components/authentication` section where you will find more
-about configuring the component, creating custom Authorization classes, and much more.
+Now you can visit ``/users/logout`` to log out. You should then be sent to the login
+page. You now have a simple blog that allows authenticated users to create or
```

Comment:
Turn this into a list to show expected accomplishments.

You now have a simple blog that allows users to perform these actions:

- Authenticated users
  - Create articles
  - Edit articles
- Unauthenticated users
  - View articles
  - View tags